filament/src/ShadowMap.cpp

```diff
@@ -453,6 +427,18 @@ void ShadowMap::computeShadowCameraDirectional(
         }
         mLightSpace = St;
 
+        // The mLightSpace matrix transforms coordinates from world space into (u, v, z)
+        // coordinates, where (u, v) are used to access the shadow map, and z is the PCF
+        // comparison value between 0 and 1.
+        // For VSM, we don't want to project the z coordinate, but do want to transform it to linear
+        // light space. We then scale the z by zfar, which prevents us from overflowing when we
+        // square z in the depth shader.
+        mLightSpaceVsm = mLightSpace;
+        mLightSpaceVsm[0][2] = Mv[0][2] / std::abs(zfar);
+        mLightSpaceVsm[1][2] = Mv[1][2] / std::abs(zfar);
+        mLightSpaceVsm[2][2] = Mv[2][2] / std::abs(zfar);
+        mLightSpaceVsm[3][2] = Mv[3][2] / std::abs(zfar);
```

Comment:
Sure, I notice we do this a lot. What's the reason?

Reply:
when you use `operator /` it's applied to all elements of the vector, so that's 4 divides for a float4, so it's a good habit. Here it's not that case, but you still have 4 divides that can be replaced 4 muls --  it wouldn't make a huge difference here, maybe not worth it, but still a good habit.